sympy/assumptions/sathandlers.py

```diff
@@ -87,21 +87,13 @@ class AllArgs(UnevaluatedOnFree):
 
     The typical usage is to evaluate predicates with expressions using .rcall().
 
-    Example
-    =======
-
-    >>> from sympy.assumptions.sathandlers import AllArgs
-    >>> from sympy import symbols, Q
-    >>> x, y = symbols('x y')
-    >>> a = AllArgs(Q.positive | Q.negative)
-    >>> a
-    AllArgs(Q.negative | Q.positive)
-    >>> a.rcall(x*y)
-    (Q.negative(x) | Q.positive(x)) & (Q.negative(y) | Q.positive(y))
     """
+    # TODO: Add examples.
 
-    def apply(self):
-        return And(*[self.pred.rcall(arg) for arg in self.expr.args])
+    def apply(self, expr=None, is_Not=False):
+        pred = to_NNF(self.pred)
+        res = AND(*[pred.rcall(arg) for arg in expr.args])
+        return ~res if is_Not else res
```

Comment:
Every apply method has this same logic in it. Can we factor that out to the superclass. I guess the subclasses should define `_eval_apply` or something like that.